File: projfabsoftescolamusica/src/main/java/br/univille/projfabsoftescolamusica/entity/Emprestimo.java
package br.univille.projfabsoftescolamusica.entity;

import jakarta.persistence.*;


import java.time.LocalDate;
import java.util.Date;
import org.springframework.format.annotation.DateTimeFormat;

@Entity
public class Emprestimo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    
    @Temporal(TemporalType.DATE)
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private Date dataInicio;

    @Temporal(TemporalType.DATE)
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private Date dataFim;

    @ManyToOne (cascade = {CascadeType.MERGE, CascadeType.REFRESH})
    private Aluno aluno;

    @ManyToOne (cascade = {CascadeType.MERGE, CascadeType.REFRESH})
    private Instrumento instrumento;

    

    // Getters e Setters
    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public Aluno getAluno() {
        return aluno;
    }

    public void setAluno(Aluno aluno) {
        this.aluno = aluno;
    }

    public Instrumento getInstrumento() {
        return instrumento;
    }

    public void setInstrumento(Instrumento instrumento) {
        this.instrumento = instrumento;
    }

    public Date getDataInicio() {
        return dataInicio;
    }

    public void setDataInicio(Date dataInicio) {
        this.dataInicio = dataInicio;
    }

    public Date getDataFim() {
        return dataFim;
    }

    public void setDataFim(Date dataFim) {
        this.dataFim = dataFim;
    }
}

File: projfabsoftescolamusica-frontend/src/app/model/emprestimo.ts
import { Aluno } from "./aluno";
import { Instrumento } from "./instrumento";

export class Emprestimo {
  id: number;
  dataInicio: string;
  dataFim: string;
  aluno: Aluno;
  instrumento: Instrumento;

  constructor(
    id: number,
    dataInicio: string,
    dataFim: string,
    aluno: Aluno,
    instrumentos: Instrumento
  ) {
    this.id = id;
    this.dataInicio = dataInicio;
    this.dataFim = dataFim;
    this.aluno = aluno;
    this.instrumento = instrumentos;
  }
}